homeassistant/components/mqtt/climate.py

```diff
@@ -447,19 +448,214 @@ async def _async_setup_entity(
     async_add_entities([MqttClimate(hass, config, config_entry, discovery_data)])
 
 
-class MqttClimate(MqttEntity, ClimateEntity):
-    """Representation of an MQTT climate device."""
+class MqttTemperatureControlEntity(MqttEntity):
+    """Helper entity class to control temperature.
 
-    _entity_id_format = climate.ENTITY_ID_FORMAT
-    _attributes_extra_blocked = MQTT_CLIMATE_ATTRIBUTES_BLOCKED
+    MqttTemperatureControlEntity supports shared methods for
+    climate and water_heater platforms.
+    """
+
+    _attr_supported_features: IntFlag
```

Comment:
Why do we declare this here?

Reply:
After making the class ABC this could be removed.